packages/resolver/src/adapters/cns-api-helpers.test.ts

```diff
@@ -0,0 +1,250 @@
+import {init} from '@emurgo/cross-csl-nodejs'
+import {Resolver} from '@yoroi/types'
+import {cnsApiConfig} from './cns-api'
+import {
+  resolveAddress,
+  resolveDomain,
+  resolveUserRecord,
+  resolveVirtualSubdomain,
+} from './cns-api-helpers'
+import {cnsCardanoApiMock} from './cns-cardano-api-maker.mocks'
+
+describe('resolveDomain', () => {
+  it('resolves an address', async () => {
+    const domain = 'fake.domain'
+
+    expect(
+      await resolveDomain(
+        domain,
+        cnsApiConfig.mainnet,
+        cnsCardanoApiMock.success,
+      ),
+    ).toBe(
+      'addr1qxjkgj7t3nvzkhsyOpjhuty0a65x26p2tvsdk3wsjkq8mp8yqjptk3jg|32t|lqxzawy48jly69hpkj5hak7f44dw80sqfylvs',
+    )
+  })
+
+  it('throws "not found" error: no metadata', async () => {
+    const domain = 'fake.domain'
+
+    try {
+      await resolveDomain(
+        domain,
+        cnsApiConfig.mainnet,
+        cnsCardanoApiMock.noMetadata,
+      )
+    } catch (e) {
+      expect(e).toBeInstanceOf(Resolver.Errors.NotFound)
+    }
+  })
```

Comment:
```suggestion
    try {
      await resolveDomain(
        domain,
        cnsApiConfig.mainnet,
        cnsCardanoApiMock.noMetadata,
      )
      fail('resolveDomain should have thrown')
    } catch (e) {
      expect(e).toBeInstanceOf(Resolver.Errors.NotFound)
    }
  })
```

or 

Otherwise, if it eventually stops throwing, it won't jump on catch and the tests will produce false positives.

```suggestion
await expect(resolveDomain(
        domain,
        cnsApiConfig.mainnet,
        cnsCardanoApiMock.noMetadata,
      ).rejects.toThrow()
  })
```